08_education/preschool_county.qmd

```diff
@@ -0,0 +1,766 @@
+
+```
+
+Confirm there are no vacant properties included in the data.
+```{r}
+acs_crosswalked %>%
+  filter(is.na(vacancy)) %>% 
+  count(pernum)
+
+```
+## Create pre-K metric: All
+
+### Pre-drop suppressed cells
+
+First, to save time when processing the metric, calculate the effective sample count for each county, defined as the sum of the `afact` variable. This is used to create a cutoff for data quality based on the actual number of survey records being used in the calculation. We will not include counties with less than 30 effective samples.
+
+```{r}
+#| label: all-effective-samples
+
+acs_all <- acs_crosswalked %>%
+  filter(sample_structure == "5year") %>% 
+  group_by(year, sample, crosswalk_period, statefip, county) %>%
+  mutate(effective_sample = sum(afact)) %>% 
+  ungroup()
+
+# calculate the number of statistics after suppressing values
+acs_all %>%
+  group_by(year, sample, crosswalk_period, statefip, county) %>%
+  summarize(
+    effective_sample = max(effective_sample)
+  ) %>%
+  ungroup() %>%
+  summarize(
+    original_stats = n(),
+    unsuppressed_stats = sum(effective_sample >= 30)
+  )
+
+```
+
+Create the preschool access metric. 
+
+Objective: calculate the percent of individuals between 3 and 4 that are enrolled in pre-K.
+
+Aggregation should be weighted by PERWT (this is a person level statistic).
+
+Remove records from counties below the size quality cutoff. 
+
+```{r}
+#| label: subset-all
+
+acs_all <- acs_all %>% 
+  filter(effective_sample >= 30)
+
+```
+
+### Calculate metric
+
+Calculate the county-level metrics. 
+
+```{r}
+#| label: calc-all
+
+results_all <- acs_all %>%
+   group_split(statefip) %>% 
+  map_dfr(~ calc_survey_mean(.data = .x, .geo_level = county))
+
+
+```
+
+Clean up the confidence interval bounds and sort the data.
+
+```{r}
+#| label: finalize-all
+
+results_all <- finalize_metric(results_all)
+
+```
+
+```{r}
+rm(acs_all)
+
+```
+
+## Add Suppressed Rows to Data
+
+To create the metric we suppressed counties that did not meet the threshold for sample size but we need to include these in the final data. Create an all version of the data that includes counties that were suppressed prior to the survey_mean calculation. 
+```{r}
+results_all_expand <- acs_crosswalked %>% 
+  filter(sample_structure == "5year") %>% 
+  mutate(geoid = paste0(statefip, county)) %>% 
+  group_by(year, geoid) %>% 
+  count() %>% 
+   expand_grid() %>%  
+   mutate(statefip = str_sub(geoid, 1, 2),
+         county = str_sub(geoid, 3, 5)) %>% 
+  left_join(results_all, by = c("year", "statefip", "county"))
+```
+
+## Data Quality Flags
+
+Add a flag for data quality, this is a numeric variable between 1 and 3 with 1 representing the best quality and 3 representing the worst. 
+
+First, join on to county_puma which provides the quality of the crosswalk (`afact * afact2`).
+```{r}
+
+metrics_education_all <- left_join(results_all_expand, county_puma, by=c("crosswalk_period", "statefip","county"))
```

Comment:
I think you can add the summarization of `geographic_allocation_quality` to `calc_survey_mean()` and `calc_survey_mean_sub()`. Be sure to use `unweighted()` so it doesn't slow down the code. 

This will simplify the code and make this step consistent across the metrics. 

Also, please document this step better. I panicked about `mean()`. Really it's just taking the value that is constant across all PUMAs within a county.

Reply:
This is a very good point and I actuallyt think the way I've been calculating the geographic_allocation_quality flag is technically wrong. The mean of quality should take into consideration the number of records coming from each PUMA. I moved into the calc_survey_mean function so this happens now.

Reply:
I don't think `geographic_allocation_quality` should vary between PUMAs within a county though.